addons/website/static/src/builder/plugins/options/social_media_option_plugin.js

```diff
@@ -172,77 +162,76 @@ class SocialMediaOptionPlugin extends Plugin {
         ],
     };
 
-    /** The social media's name for which there is an entry in the orm */
-    async getRecordedSocialMediaNames() {
-        await this.fetchRecordedSocialMedia();
-        return this.recordedSocialMedia.keys();
-    }
-
-    // TODO: a method to give access to the `recordedSocialMedia` for facebook page and instagram page
+    /**
+     * Fetches social media links from the company record and pre-fills the
+     * corresponding links in the snippet, replacing internal
+     * "/website/social/<name>" entries.
+     */
+    async prefillSocialMediaLinks(snippetEl) {
+        this.companySocialLinks = new Map();
+        const socialMediaInfoFields = [
+            ...socialMediaInfo
+                .entries()
+                .filter(([name, info]) => info.recorded)
+                .map(([name, info]) => `social_${name}`),
+        ];
 
-    setup() {
-        this.recordedSocialMedia = new Map();
-    }
+        const websiteId = this.services.website?.currentWebsite?.id;
+        const websiteRes = await this.services.orm.read("website", [websiteId], ["company_id"]);
+        const companyId = websiteRes?.[0]?.company_id?.[0];
```

Comment:
I believe the website service is available throughout, could you please check once again 🤔